app/models/audit/event/whoami.rb

```diff
@@ -0,0 +1,83 @@
+module Audit
+  module Event
+    class Whoami
+
+      # NOTE: "role" may refer to a user or host.
+      def initialize(
+        client_ip:,
+        role:,
+        success:
+      )
+        @client_ip = client_ip
+        @role = role
+        @success = success
+
+        # Implements `==` for audit events
+        @comparable_evt = ComparableEvent.new(self)
+      end
+
+      # NOTE: We want this class to be responsible for providing `progname`.
+      # At the same time, `progname` is currently always "conjur" and this is
+      # unlikely to change.  Moving `progname` into the constructor now
+      # feels like premature optimization, so we ignore reek here.
+      # :reek:UtilityFunction
+      def progname
+        Event.progname
+      end
+
+      def severity
+        attempted_action.severity
+      end
+
+      def to_s
+        message
+      end
+
+      def message
+        "#{role_id} checked identity using WhoAmI"
```

Comment:
i still think we should change this to `"#{role_id} checked its identity using whoami"`

Reply:
@netacoral @shulifink

Reply:
updated and reviewed by both